test/integration/template_database_test.go

```diff
@@ -1787,3 +1800,126 @@ func TestDatabaseTopologyConstraints(t *testing.T) {
 		verifyTopologyConstraints(obj.Name, obj.Spec.Template.Spec, expectedLabels)
 	}
 }
+
+func TestDatabaseStorageGroups(t *testing.T) {
+	t.Run("testStorageGroupEnabled", func(t *testing.T) {
+		// Path to the helm chart we will test
+		helmChartPath := testlib.DATABASE_HELM_CHART_PATH
+		options := &helm.Options{
+			SetValues: map[string]string{
+				"database.sm.nohotCopy.replicas":   "1",
+				"database.sm.storageGroup.enabled": "true",
+			},
+		}
+
+		// SGs are passed to nuosm
+		output := helm.RenderTemplate(t, options, helmChartPath,
+			"SG1", []string{"templates/statefulset.yaml"})
+		for _, obj := range testlib.SplitAndRenderStatefulSet(t, output, 2) {
+			args := obj.Spec.Template.Spec.Containers[0].Args
+			assert.True(t, testlib.ArgContains(args, "--storage-groups"))
+			assert.True(t, testlib.ArgContains(args, "SG1"))
+		}
+	})
+
+	t.Run("testStorageGroupWithName", func(t *testing.T) {
+		// Path to the helm chart we will test
+		helmChartPath := testlib.DATABASE_HELM_CHART_PATH
+		options := &helm.Options{
+			SetValues: map[string]string{
+				"database.sm.nohotCopy.replicas":   "1",
+				"database.sm.storageGroup.enabled": "true",
+				"database.sm.storageGroup.name":    "SG1",
```

Comment:
Should you pick a SG name that is not the default?

Reply:
The default SG name will be the Helm release name which is `release-name` in this test.